transcom/mymove
# INT-B-21618 Fix NTS/NTSR Creation for Prime

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1027358)

## Summary
The purpose of this story is to fix the NTS/NTSR Creation for the prime user. Previously it was erroring if the user sent a delivery address(NTS) or a pickup address(NTSR). The reason was that the shipment creator service was skipping the address creation in the database for NTS and NTSR which happens before the shipment is saved. Subsequently when it attempted to save the shipment, it was encountering a FK constraint on the address due to the address not being saved to the database.

This story removes the NTS/NTSR shipment type check for delivery/pickup address creation which will allow the shipment to be created successfully. The current process of the TOO adding the storage facility address info and that info being copied to the destination(NTS) or pickup(NTSR) address remains ensuring that the storage facility address will match the pickup/destination address as expected.

There will be a follow on feature that will allow the prime to include the storage information when creating an NTS/NTSR shipment, but that was not included as part of this feature.

Is there anything you would like reviewers to give additional scrutiny?

[this article](tbd) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Login as a prime simulator user.
2. Choose an existing move.
3. Click on the Create Shipment button.
4. Choose either HHG_INTO_NTS or HHT_OUTOF_NTS_DOMESTIC
5. Add a pickup address, destination address, and any other required fields.
6. Click the Save button. You should received a "Successfully created shipment..."

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
![Screenshot 2024-12-27 at 12 32 11 PM](https://github.com/user-attachments/assets/c20d70c6-c15e-4ca0-9d2d-cbf4eca076d1)
![Screenshot 2024-12-27 at 12 32 33 PM](https://github.com/user-attachments/assets/2652288a-eeb4-4a88-b973-a69dac0fc8db)
![Screenshot 2024-12-27 at 12 33 08 PM](https://github.com/user-attachments/assets/6ac46c99-2bd0-44cb-acb0-37c29b4259a0)
